tests/unit_tests.rs

```diff
@@ -615,10 +642,45 @@ fn test_read_write_roundtrip_escape() {
         match reader.read_event(&mut buf) {
             Ok(Eof) => break,
             Ok(Text(e)) => {
-                let t = e.unescaped().unwrap();
+                let t = e.escaped();
+                assert!(
+                    writer
+                        .write_event(Event::Text(BytesText::owned(t.into_owned())))
+                        .is_ok()
+                );
+            }
+            Ok(e) => assert!(writer.write_event(e).is_ok()),
+            Err(e) => panic!(e),
+        }
+    }
+
+    let result = writer.into_inner().into_inner();
+    assert_eq!(String::from_utf8(result).unwrap(), input.to_string());
+}
+
+#[test]
+fn test_read_write_roundtrip_escape_text() {
+    let input = r#"
+        <?xml version="1.0" encoding="UTF-8"?>
+        <section ns:label="header">
+            <section ns:label="empty element section" />
+            <section ns:label="start/end section"></section>
+            <section ns:label="with text">data &lt;escaped&gt;</section>
+            </section>
+    "#;
+
+    let mut reader = Reader::from_str(input);
+    reader.trim_text(false).expand_empty_elements(false);
+    let mut writer = Writer::new(Cursor::new(Vec::new()));
+    let mut buf = Vec::new();
+    loop {
+        match reader.read_event(&mut buf) {
+            Ok(Eof) => break,
+            Ok(Text(e)) => {
+                let t = String::from_utf8(e.unescaped().unwrap().into_owned()).unwrap();
```

Comment:
I'd prefer using a `unescape_and_decode` or `unescape` then `decode` separately instead of `String::from_utf8`. Could you change it?